[GraphBolt] Fixing two issues in GraphBolt tests.

drivanov opened the pull request:
## Description
<!-- Brief description. Refer to the related issues if existed.
It'll be great if relevant reviewers can be assigned as well.-->
The first issue we fixed here is the following warning appearing in `test_basic_feature_store.py::test_basic_feature_store_errors` 
```
tests/python/pytorch/graphbolt/impl/test_basic_feature_store.py::test_basic_feature_store_errors
  /usr/local/lib/python3.10/dist-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function TorchBasedFeature.__del__ at 0x7f282dcad6c0>
 
  Traceback (most recent call last):
    File "/usr/local/lib/python3.10/dist-packages/dgl/graphbolt/impl/torch_based_feature_store.py", line 93, in __del__
      for tensor in self._is_inplace_pinned:
  AttributeError: 'TorchBasedFeature' object has no attribute '_is_inplace_pinned'
 
    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
```


The second issue is a bug that occurs in the `test_ondisk_dataset.py::test_OnDiskDataset_heterogeneous` test, causing the following warning:
```
tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py::test_OnDiskDataset_heterogeneous[csv-True]
tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py::test_OnDiskDataset_heterogeneous[csv-False]
tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py::test_OnDiskDataset_heterogeneous[numpy-True]
tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py::test_OnDiskDataset_heterogeneous[numpy-False]
  /opt/dgl/dgl-source/tests/python/pytorch/graphbolt/gb_test_utils.py:272: UserWarning: you are shuffling a 'Tensor' object which is not a subclass of 'Sequence'; `shuffle` is not guaranteed to behave correctly. E.g., non-numpy array/tensor objects with view semantics may contain duplicates after shuffling.
    np.random.shuffle(user_ids)
```

In the debugger, I see incorrect shuffle results for `user_ids` which has no duplicates before shuffling:
```python
(Pdb) user_ids
tensor([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,
         14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,  26,  27,
         28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,  39,  40,  41,
         42,  43,  44,  45,  46,  47,  48,  49,  50,  51,  52,  53,  54,  55,
         56,  57,  58,  59,  60,  61,  62,  63,  64,  65,  66,  67,  68,  69,
```
After shuffling there are a lot of them: 
```python
(Pdb) user_ids
tensor([  0,   1,   1,   3,   0,   2,   1,   3,   8,   5,   4,   9,  10,   1,
          9,   4,   6,   8,   9,   3,   4,  13,  14,   5,   8,   6,  16,  21,
         13,  19,  15,   3,  20,  17,  13,  10,   5,   7,  12,  25,   0,  18,
         39,  28,  12,  26,  43,   0,  38,  19,  31,  50,  38,  40,  16,   1,
         48,  40,  36,  57,  19,   2,  43,   5,  35,  28,  51,   2,  66,  37,
         31,  21,  65,  27,  68,  25,  25,  44,  32,  26,  30,  13,  20,   6,
``` 
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change


## Changes
<!-- You could use following template
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)
-->

mfbalin commented:
One of the warning was fixed in #7044. If you merge into the latest master, I believe it should go away.

mfbalin commented on tests/python/pytorch/graphbolt/impl/test_basic_feature_store.py:
```diff
@@ -91,6 +91,7 @@ def test_basic_feature_store_hetero():
     ]
 
 
+@pytest.mark.filterwarnings("ignore::pytest.PytestUnraisableExceptionWarning")
```
Is this change still needed after merging into master?

Rhett-Ying commented:
@dgl-bot

Rhett-Ying reviewed (commented):
please verify if `is_inplace_pinned` and revert the change here. others look good to me. please ping me once addressed the comments.

drivanov commented:
> please verify if `is_inplace_pinned` and revert the change here. others look good to me. please ping me once addressed the comments.

@Rhett-Ying :  just checked out [fix#7044](https://github.com/dmlc/dgl/pull/7044) for `is_inplace_pinned` and reverted my change for this issue.

Rhett-Ying commented:
@dgl-bot
